test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -1290,4 +1336,100 @@ TEST_P(ExtProcIntegrationTest, BufferBodyOverridePostWithRequestBody) {
   verifyDownstreamResponse(*response, 200);
 }
 
+// Set up per-route configuration that sets a custom processing mode on the
+// route, and test that the processing mode takes effect.
+TEST_P(ExtProcIntegrationTest, PerRouteProcessingMode) {
+  initializeConfig();
+  modifyHttpVirtualHost([this](VirtualHost& vh) {
+    // Set up "/foo" so that it will send a buffered body
+    auto* route = vh.mutable_routes()->Mutable(0);
+    route->mutable_match()->set_path("/foo");
+    ExtProcPerRoute per_route;
+    per_route.mutable_overrides()->mutable_processing_mode()->set_response_body_mode(
+        ProcessingMode::BUFFERED);
+    setPerRouteConfig(route, per_route);
+  });
+  HttpIntegrationTest::initialize();
+
+  auto response =
+      sendDownstreamRequest([](Http::RequestHeaderMap& headers) { headers.setPath("/foo"); });
+  processRequestHeadersMessage(true, absl::nullopt);
+  Buffer::OwnedImpl full_response;
+  TestUtility::feedBufferWithRandomCharacters(full_response, 100);
+  handleUpstreamRequestWithResponse(full_response, 100);
+  processResponseHeadersMessage(false, absl::nullopt);
+  // Because of the per-route config we should get a buffered response
+  processResponseBodyMessage(false, [&full_response](const HttpBody& body, BodyResponse&) {
+    EXPECT_TRUE(body.end_of_stream());
+    EXPECT_EQ(body.body(), full_response.toString());
+    return true;
+  });
+  verifyDownstreamResponse(*response, 200);
+}
+
+// Set up configuration on the virtual host and on the route and see that
+// the two are merged.
+TEST_P(ExtProcIntegrationTest, PerRouteAndHostProcessingMode) {
+  Logger::Registry::getLog(Logger::Id::filter).set_level(spdlog::level::trace);
```

Comment:
is this used for debug?

Reply:
Sorry, mistakenly left in. (Ask me sometime why I do that and not rely on the command line arguments ;-)